### PR TITLE
Add cluster support for ClickHouse

### PR DESCRIPTION
**Pull request by srikanthccv**
Currently, the completed migration only inserts the data into the shard it is connected to, causing migration to run again when it is connected to a different shard. 

We want to insert the same data into all shards of all replicas. This can't be achieved with distributed table engine because it requires sharding data which is not desired because we want same data in all shards not some randomly distributed data.

To achieve this we make use of `INSERT INTO FUNCTION remote ...` https://clickhouse.com/docs/en/sql-reference/statements/insert-into#inserting-using-a-table-function. This inserts the data into all other instances of ClickHouse shard/replica. The hosts are obtained by querying the `system.clusters` table. However, this table contains one entry with `localhost` for the shard we are connected to, so we ignore that entry and then write to connected shard separately.

## Cons of this approach

You can't run from anywhere out the n/w of ClickHouse instance running ex (you can't run from a local machine). Should I make it configurable where it tries to write to remote instances?

### CODE REVIEW

File: database/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -178,6 +178,24 @@ func (ch *ClickHouse) Version() (int, bool, error) {
 	return version, dirty == 1, nil
 }
 
+func (ch *ClickHouse) HostAddrs() ([]string, error) {
+	var hostAddrs []string
+	query := "SELECT DISTINCT host_address FROM system.clusters WHERE host_address NOT IN ['localhost', '127.0.0.1'] AND cluster = '" + ch.config.ClusterName + "'"
```
Comment (nityanandagohain) [inline database/clickhouse/clickhouse.go:194]: when I run this query on clickhouse running in docker
```
SELECT DISTINCT host_address FROM system.clusters

SELECT DISTINCT host_address
FROM system.clusters

Query id: 4c4182eb-483e-46a2-aafc-b46395ea9bbb

┌─host_address─┐
│ 172.22.0.3   │
└──────────────┘
````
So here the ip is `172.22.0.3` and it won't be excluded and the data will be inserted twice to the same cluster right ?

**Review by nityanandagohain — COMMENTED** [review]
overall the changes looks good, but have this small doub't

File: database/clickhouse/clickhouse.go (hunk 1 above)
Comment (srikanthccv) [reply database/clickhouse/clickhouse.go:194]: I haven't tested with the docker. I will take a look.

Comment (srikanthccv) [reply database/clickhouse/clickhouse.go:194]: I updated the logic. Now, it fetches the host addresses from the shard/host it is connected excluding its own `localhost/127.0.0.1`. Then, it connects to one of the other hosts and does the same thing to get the host address of the first host we connected to. This gives us a unique list of host addresses to insert into.

Comment (srikanthccv) [reply database/clickhouse/clickhouse.go:194]: This logic should work for any env

**Review by nityanandagohain — APPROVED** [review]

**Pull request merged**